lib/workers/repository/process/index.ts

```diff
@@ -1,24 +1,53 @@
 import { mergeChildConfig } from '../../../config';
+import { GlobalConfig } from '../../../config/global';
 import type { RenovateConfig } from '../../../config/types';
 import { logger } from '../../../logger';
 import type { PackageFile } from '../../../manager/types';
+import { platform } from '../../../platform';
+import { getCache } from '../../../util/cache/repository';
 import { branchExists } from '../../../util/git';
 import { addSplit } from '../../../util/split';
 import type { BranchConfig } from '../../types';
 import { readDashboardBody } from '../dependency-dashboard';
 import { ExtractResult, extract, lookup, update } from './extract-update';
 import type { WriteUpdateResult } from './write';
 
-function getBaseBranchConfig(
+async function getBaseBranchConfig(
   baseBranch: string,
   config: RenovateConfig
-): RenovateConfig {
+): Promise<RenovateConfig> {
   logger.debug(`baseBranch: ${baseBranch}`);
-  const baseBranchConfig = mergeChildConfig(config, { baseBranch });
-  if (config.baseBranches.length > 1) {
+
+  let baseBranchConfig: RenovateConfig = config;
+
+  if (
+    config.useBaseBranchConfig === 'replace' &&
+    baseBranch !== config.defaultBranch
+  ) {
+    logger.debug(
+      `Using config from branch ${baseBranch} because useBaseBranchConfig option specified`
+    );
+
+    // Retrieve config file name autodetected for this repo
+    const cache = getCache();
+    const { configFileName } = cache;
+
+    baseBranchConfig = await platform.getJsonFile(
+      configFileName || config.onboardingConfigFileName,
+      config.repository,
+      baseBranch
+    );
+
+    baseBranchConfig = mergeChildConfig(GlobalConfig.get(), baseBranchConfig);
```

Comment:
I don't think you want to merge the global config here, as those values shouldn't be on branch config. 🤔 
```suggestion
    baseBranchConfig = mergeChildConfig(config, baseBranchConfig);
```